src/env.h

```diff
@@ -781,6 +781,9 @@ class Environment : public MemoryRetainer {
   void stop_sub_worker_contexts();
   template <typename Fn>
   inline void ForEachWorker(Fn&& iterator);
+  // Determine if the environment is stopping. This getter is thread-safe.
+  // If the check is not performed off thread, the flag
+  // `env->can_call_into_js()` should be preferred.
```

Comment:
I wouldn't suggest doing this. Even if the flags are currently set in similar situations, they have different semantics; they are not replacements for each other.